internal/metrics/recorder.go

```diff
@@ -0,0 +1,62 @@
+package metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/fluxcd/pkg/apis/meta"
+)
+
+const (
+	ConditionDeleted = "Deleted"
+)
+
+type Recorder struct {
+	readyGauge *prometheus.GaugeVec
+}
+
+func NewRecorder() *Recorder {
+	return &Recorder{
+		readyGauge: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: "",
+				Subsystem: "",
+				Name:      "gitops_toolkit_ready_condition",
```

Comment:
Are we sure we don't want to parameterize this metric on the condition type? Even if it is decided not to expose the other condition types yet, this would at least allow for that possibility in the future. Which would then be useful for building generic dashboards for all condition types without having to hard code each separate metric name for each condition type.

Reply:
We could make it `gitops_toolkit_condition` with the a label `type` or you had something different in mind?

Reply:
That works for me. If we want something more specific could be e.g. `gitops_toolkit_condition_status_equals`.